runtime/opt/taupage/init.d/04-configure-network-interfaces.py

```diff
@@ -0,0 +1,226 @@
+
+import boto.ec2
+import boto.utils
+import time
+import sys
+import subprocess
+import netifaces
+from netaddr import IPAddress
+
+from taupage import configure_logging, get_config
+
+
+def instance_id():
+    """Helper to return theid for the current instance"""
+    return boto.utils.get_instance_metadata()['instance-id']
+
+
+def detect_region():
+    """Helper to return the region for the current instance"""
+    return boto.utils.get_instance_metadata()['placement']['availability-zone'][:-1]
+
+
+def zone():
+    """Helper to return the AZ for the current instance"""
+    return boto.utils.get_instance_metadata()['placement']['availability-zone']
+
+
+def retry(func):
+    def wrapped(*args, **kwargs):
+        count = 0
+        while True:
+            try:
+                return func(*args, **kwargs)
+            except boto.exception.BotoServerError as e:
+                if count >= 10 or str(e.error_code) not in ('Throttling', 'RequestLimitExceeded'):
+                    raise
+                logging.info('Throttling AWS API requests...')
+                time.sleep(2 ** count * 0.5)
+                count += 1
+
+    return wrapped
+
+
+class CmdException(Exception):
+    def __init__(self, returncode, errmsg):
+        self._returncode = returncode
+        message = "Command returned non-zero exit status {}:\n{}".format(returncode, errmsg)
+        super(CmdException, self).__init__(message)
+
+    @property
+    def returncode(self):
+        return self._returncode
+
+
+def call_command(call, allowed_error_codes=[0]):
+    proc = subprocess.Popen(call, stderr=subprocess.PIPE)
+    stdout, stderr = proc.communicate()
+    if proc.returncode not in allowed_error_codes:
+        raise CmdException(proc.returncode, stderr.decode('utf-8'))
+    return stdout
+
+
+def ec2_client(region):
+    return boto.ec2.connect_to_region(region)
+
+
+@retry
+def get_all_network_interfaces(ec2, filters):
+    return ec2.get_all_network_interfaces(filters=filters)
+
+
+def find_network_interface(ec2, name):
+    tries = 10
+    network_interfaces = []
+    while not network_interfaces:
+        try:
+            network_interfaces = list(get_all_network_interfaces(ec2, {
+                'tag:Name': name,
+                'status': 'available',
+                'availability-zone': zone()
+            }))
+        except Exception as e:
+            logging.exception(e)
+            sys.exit(2)
+
+        if not network_interfaces:
+            logging.error(
+                'No matching "available" network interfaces volume with name %s found.', name)
+            tries -= 1
+            if tries > 0:
+                logging.error(
+                    'Sleeping for 10 seconds and hope a network interface will become "available"')
+                time.sleep(10)
+            else:
+                sys.exit(2)
+
+    if len(network_interfaces) > 1:
+        logging.warning('More than one network interface with name %s found.', name)
```

Comment:
this doesn't need to be a warning imho

Reply:
I would consider it a warning, could result in the instance getting an unexpected IP.

Should two interfaces exist within the same AZ, they have the same name and are both available. The instance will attempt to attach the head.